bittensor/_dataset/dataset_impl.py

```diff
@@ -1,678 +1,772 @@
-        self.refresh_corpus = True
+    def construct_text_corpus(self, datasets:Optional[List[str]] = None, save:bool = False, load:bool = False) -> None :
+        """ Building all of the datasets specified by getting each of their 
+            text hashes from IPFS or local
+        Args:
+            datasets (List[str], optional):
+                List of dataset names to include from the pile.
+            save (bool, required):
+                Save the dataset hashes locally.
+            load (bool, required):
+                Load the dataset hashes locally.
+        """
 
-        self.build_hash_table()
+        datasets = datasets if datasets else self.datasets
 
-        os.makedirs(os.path.expanduser(data_dir), exist_ok=True)
-            
-        self.data_queue = ThreadQueue(
-            producer_target = self.reserve_multiple_data,
-            producer_arg = (self.num_batches, ),
-            buffer_size = 1
-        )
+        all_text_file_metas = []
+        dataset_hash_map = {}
 
-    def __del__(self):
-        self.close()
+        tasks = []
 
-    def close(self):
-        self.data_queue.close()
+        self.dataset_size_map = {d:0 for d in datasets}
+        
+        # Gather dataset hashes async as their state is independent.
+        for dataset in datasets:
+            tasks += [self.async_build_single_dataset(dataset=dataset, save=save, load=load)]
 
-    def get_folder_size(self, folder):
-        r""" Get the size (in byte) of a folder inside the data_dir.
-        Args:
-            folder (str):
-                The name of the folder
+        # Get the hashes asynchronously for each dataset.
+        dataset_hashes = asyncio.run(asyncio.gather(*tasks))
+
+        # Create a hash map of dataset -> text hashes.
+        for k,v in zip(datasets, dataset_hashes):
+            if len(v) > 0:
+                dataset_hash_map[k] = v
         
-        Returns:
-            total_size (int):
-                The memory size of the folder (in byte). 
-        """
-        total_size = 0
-        full_path = os.path.expanduser(os.path.join(self.data_dir, folder))
-        for dirpath, dirnames, filenames in os.walk(full_path):
-            for f in filenames:
-                fp = os.path.join(dirpath, f)
-                # skip if it is symbolic link
-                if not os.path.islink(fp):
-                    total_size += os.path.getsize(fp)
-
-        return total_size
-
-    def load_hash(self, file_meta):
-        r""" Load a hash from disk.
-        Args:
-            file_meta (dict of str: int):
-                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
+        self.dataset_hash_map = dataset_hash_map
 
-        Returns:
-            text (str): 
-                The text in the file.                
+        # Flatten the hashes to a list of hashes.
+        for  k,file_meta_list in dataset_hash_map.items():
```

Comment:
Some strange spacing here. Generally, there should not be double spaces between variables nor extra lines. If you need an extra line it is a good opportunity to insert a comment.

Reply:
I get the double spaces, what do you mean by extra lines? Like double lines between code lines?